website/static/js/components/legend.js

```diff
@@ -1,17 +1,56 @@
 require('css/legend.css');
 var m = require('mithril');
 
+/** Finds the index of a string in an array
+ *
+ * @param {Array} array The array to search through
+ * @param {String} str The string to find
+ * @returns {*}
+ */
+var find_index = function (array, str) {
+    for (var i = 0; i < array.length; i++) {
+        if (array[i].label === str)
+            return i;
+    }
+    return -1;
+};
+
+/** Removes an index from an array immutably
+ *
+ * @param {Array} array the array to modify
+ * @param {Int} index the index to remove
+ * @returns {*} a new array missing the index value
+ */
+var remove_index = function (array, index) {
+    var new_array = [];
+    for (var i = 0; i < array.length; i++) {
+        if (i !== index)
+            new_array.push(array[i]);
+    }
+    return new_array;
+};
+
 module.exports = {
-    view: function(data, repr, opts) {
-        if(data[0].label){
-            data.sort(function(a, b) {
+    view: function (data, repr, opts) {
+        var other_index = find_index(data, 'Other');
```

Comment:
Most of it is coming from the `OrderedDict` in `website/project/model.py` that you modified, but it's being extended in `website/static/js/projectorganizer.js`, which breaks ordering. My bad.

This file could be simplified by modifying the sort function instead of the data, which would also give a small performance boost.